### PR TITLE
Implemented unit tests for strstr() and strcasestr()

### LINKED ISSUE #1855: Add unit test for strstr() & strcasestr()
Add some test cases for standard strstr() & strcasestr()

### PR DESCRIPTION
**Pull request by VaibKS**
Closes #1855

### CODE REVIEW

File: src/tests/stdlib/strcasestr.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+#include <string.h>
+#include <embox/test.h>
+
+EMBOX_TEST_SUITE("strcasestr test suite");
+
+TEST_CASE("strcasestr with no needle in the haystack (no match)") {
+	test_assert_zero(strstr("the quick brown fox", "wolf"))
+	test_assert_zero(strstr("the quick brown fox", "wolf"))
```
Comment (anton-bondarev) [inline src/tests/stdlib/strcasestr.c]: 2 identical string?

File: src/tests/stdlib/Mybuild [hunk 2]
```diff
@@ -123,3 +123,16 @@ module memcmp_test {
...
+	depends embox.compat.libc.str
+	depends embox.framework.LibFramework
+}
+
+module strcasestr_test {
+	source "strcasestr.c"
+
+	depends embox.compat.libc.str
+	depends embox.framework.LibFramework
+}
```
Comment (anton-bondarev) [inline src/tests/stdlib/Mybuild]: a new line please

Comment (VaibKS) [reply src/tests/stdlib/Mybuild]: Yes, will fix both

File: src/tests/stdlib/strcasestr.c (hunk 1 above)
Comment (VaibKS) [reply src/tests/stdlib/strcasestr.c]: Sorry for that. Fixing it.

**Comment by anton-bondarev** [comment]
@VaibKS Thanks
Can you also add some more tests
For example
There are 2 suit substring in string, check that it a pointer to first substring

**Comment by VaibKS** [comment]
Yes, will do that.

**Comment by anton-bondarev** [comment]
@VaibKS Thanks
Now rebase and squash commits
Then add your tests to x86/test/unit template by a separate commit

**Comment by VaibKS** [comment]
Sorry about that, issue with my text editor adding a newline when there wasn't a newline in the file. Fixed it and pushed it.

**Comment by VaibKS** [comment]
Done!

**Pull request merged**